vault/capabilities.go

```diff
@@ -0,0 +1,46 @@
+package vault
+
+import "fmt"
+
+// Capabilities is used to fetch the capabilities of the given token on the given path
+func (c *Core) Capabilities(token, path string) ([]string, error) {
+	if path == "" {
+		return nil, fmt.Errorf("missing path")
+	}
+
+	if token == "" {
+		return nil, fmt.Errorf("missing token")
+	}
+
+	te, err := c.tokenStore.Lookup(token)
+	if err != nil {
+		return nil, err
+	}
+	if te == nil {
+		return nil, fmt.Errorf("invalid token")
```

Comment:
I think this should either return DenyCapability, or we should return two potential errors, so that this can be returned as an invalid request. Same with above. An alternative might be to use go-errwrap to be able to figure out what type of error it is from the http handler and return the appropriate status code.